pkg/clusterregistry/options/options.go

```diff
@@ -28,29 +28,35 @@ import (
 
 // Runtime options for the clusterregistry-apiserver.
 type ServerRunOptions struct {
-	GenericServerRunOptions *genericoptions.ServerRunOptions
-	Etcd                    *genericoptions.EtcdOptions
-	SecureServing           *genericoptions.SecureServingOptions
-	Audit                   *genericoptions.AuditOptions
-	Features                *genericoptions.FeatureOptions
-	Authentication          *BuiltInAuthenticationOptions
+	GenericServerRunOptions  *genericoptions.ServerRunOptions
+	Etcd                     *genericoptions.EtcdOptions
+	SecureServing            *genericoptions.SecureServingOptions
+	Audit                    *genericoptions.AuditOptions
+	Features                 *genericoptions.FeatureOptions
+	StandaloneAuthentication *BuiltInAuthenticationOptions
+	DelegatedAuthentication  *genericoptions.DelegatingAuthenticationOptions
+	DelegatedAuthorization   *genericoptions.DelegatingAuthorizationOptions
 
 	EventTTL time.Duration
+	// StandaloneMode - if true, does not depend on a kube-apiserver
+	StandaloneMode bool
 }
 
 // NewServerRunOptions creates a new ServerRunOptions object with default values.
 func NewServerRunOptions() *ServerRunOptions {
 	o := &ServerRunOptions{
 		GenericServerRunOptions: genericoptions.NewServerRunOptions(),
-		Etcd:           genericoptions.NewEtcdOptions(storagebackend.NewDefaultConfig("/registry/clusterregistry.kubernetes.io", nil)),
-		SecureServing:  genericoptions.NewSecureServingOptions(),
-		Audit:          genericoptions.NewAuditOptions(),
-		Features:       genericoptions.NewFeatureOptions(),
-		Authentication: NewBuiltInAuthenticationOptions().WithAll(),
+		Etcd:                     genericoptions.NewEtcdOptions(storagebackend.NewDefaultConfig("/registry/clusterregistry.kubernetes.io", nil)),
+		SecureServing:            genericoptions.NewSecureServingOptions(),
+		Audit:                    genericoptions.NewAuditOptions(),
+		Features:                 genericoptions.NewFeatureOptions(),
+		StandaloneAuthentication: NewBuiltInAuthenticationOptions().WithAll(),
+		DelegatedAuthentication:  genericoptions.NewDelegatingAuthenticationOptions(),
+		DelegatedAuthorization:   genericoptions.NewDelegatingAuthorizationOptions(),
```

Comment:
Instead of adding the options and then having to deal with the conflicts, I would suggest doing the following so that we can get aggregation working quickly:

- don’t add options to cli for now
- add only the boolean switch to enable aggregation
- when aggregation is turned on, just use the defaukt aggregation options

AFAICT, in the service-catalog helm chart, we do not actually set any of the auth-specific CLI flags and aggregation works just fine with the default options.